libs/x-content/src/main/java/org/elasticsearch/xcontent/support/filtering/FilterPath.java

```diff
@@ -71,6 +71,17 @@ private boolean isFinalNode() {
      * @return true if the name equal a final node, otherwise return false
      */
     boolean matches(String name, List<FilterPath> nextFilters) {
+        return matches(name, nextFilters, false);
```

Comment:
I don't think we need to keep the two argument version of this method.

Reply:
Exactly! I did this because I want to make the changes as smaller as possible. I will change it.